crates/rmf_site_editor/src/site/save.rs

```diff
@@ -1725,3 +1763,149 @@ pub fn save_site(world: &mut World) {
+            .join("test_original.site.json");
+        let destination = target_test_dir
+            .join(test_site_dir)
+            .join("test_destination.site.json");
+
+        std::fs::create_dir_all(target_test_dir.join(test_site_dir)).unwrap();
+
+        // Copy the source file into the test directory.
+        //
+        // Later we will do a diff between the two files to make sure they are
+        // exactly equal, but saving to a new folder can alter the relative
+        // paths within the site. To make sure the exported copy is exactly the
+        // same as the original, the copy must be saved to the same folder as
+        // the original. However we do not want tests to produce files in a
+        // source folder, so we copy the source file into the target directory
+        // and then do a roundtrip into the same directory.
+        std::fs::copy(&source, &original).unwrap();
+
+        #[cfg(unix)]
+        {
+            // Create a symlink to avoid a slew of error log messages while loading.
+            // We can ignore the result of this, because the test should still pass
+            // even if the symlinking doesn't work, we'll just see some noisy error
+            // logs in stdout.
+            let _ = std::os::unix::fs::symlink(
+                assets_dir.join("models"),
+                target_test_dir.join("models"),
+            );
+            let _ = std::os::unix::fs::symlink(
+                assets_dir.join("drawings"),
+                target_test_dir.join("drawings"),
+            );
+        }
+
+        let destination = destination.to_str().unwrap().to_owned();
+
+        let mut app = App::new();
+        app.insert_resource(Autoload::file(original.clone(), None))
+            .add_plugins(SiteEditor::default().save_as_path(Some(destination.clone())))
+            .add_plugins(TestTimeoutPlugin::new(Duration::from_secs(10)));
+
+        // Run until the file is saved or until the timeout occurs
+        app.run();
+
+        assert!(std::fs::exists(&destination).unwrap());
+
+        #[cfg(not(target_os = "windows"))]
+        {
+            // For non-windows we can just compare the new and old files directly
+            let original = original.to_str().unwrap().to_owned();
+            assert!(file_diff::diff(&original, &destination));
+
+            let source = source.to_str().unwrap().to_owned();
+            assert!(file_diff::diff(&source, &destination));
+        }
+
+        #[cfg(target_os = "windows")]
+        {
+            use std::os::windows::prelude::*;
+            // Windows uses different characters to represent newlines and path
+            // separators, so we cannot do a 1-to-1 comparison between the original
+            // and the generated file. We will simply check whether the new file
+            // is within 10% the size of the original file size since the differences
+            // incurred by these format changes should not be too significant.
```

Comment:
I wonder, if we parsed both original and saved file into a string and:

* Did a search and replace that does "\r\n" -> "", followed by one that does "\n" -> "" for newlines and
* Did a search and replaces that does "\" -> "/" for path separators (a bit uglier)

Would that capture all cases and still allow us to have a more meaningful test? I'm not particularly worried about CI since we run both Windows and Linux and if something is wrong in the saving logic the Linux CI will probably catch it, it's more for local development for Windows users that they might not catch it.